### ISSUE TITLE
Video support for studies

### ISSUE THREAD

**Issue by Siderite** [Siderite/lichessTools#626]
So the Chessmood site (and probably many others) have a video, followed by some puzzles based on that video. Since Lichess does allow YouTube videos (hopefully in study pages as well) then it would be great if we could encode "video chapters"

**Comment by Siderite** [comment]
Suggestion:
- replace URLs in the embed format (https://www.youtube.com/embed/AaV3Oun5PDE?autoplay=1&autohide=0&showinfo=0&rel=0&disablekb=1) which allows creators to specify parameters
- the fullscreen option will be disabled if possible, since lichess doesn't allow it
- the origin and widget_referrer will be set automatically by the code
- what about autoplay?
- where to replace the URL (not everywhere in Lichess, right?)
- take into account a study you contribute to, Preview mode and study you don't contribute to and non interactive chapter
- take into account desktop and mobile
- what's the actual flow: show board and video? video replaces board? Can you start a video and then continue to other moves with it picture-in-picture? do you show the image in the interactive bubble? Maybe a video per chapter or too restrictive?

More about parameters: https://developers.google.com/youtube/player_parameters

**Comment by Siderite** [comment]
Example of a working iframe:
```
<iframe 
  title="Bobby Fischer's 21-move brilliancy" 
  width="100%"
  height="100%" 
  frameborder="0" 
  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" 
  allowfullscreen=""
  src="https://www.youtube.com/embed/AaV3Oun5PDE?autoplay=1&amp;state=1&amp;start=2&amp;autoplay=1&amp;autohide=0&amp;showinfo=0&amp;rel=0"
  credentialless=""
/>

```

**Comment by Siderite** [comment]
What about enablejsapi? Could a video be connected to specific moves at different times or something?

**Comment by Siderite** [comment]
I am thinking more of showing the video as picture in picture when clicking the link and not opening it in new window or tab. If a video is already running, close and start this one.

**Comment by Siderite** [comment]
Implemented in v2.3.113
